referrer-policy/generic/referrer-policy-test-case.js

```diff
@@ -0,0 +1,144 @@
+var ReferrerPolicyTestCase = function(scenario, testDescription) {
+  // Check if scenario is valid.
+  test(function() {
+    // We extend the exsiting test_expansion_schema not to kill performance by
+    // copying.
+    var expectedFields = SPEC_JSON["test_expansion_schema"];
+    expectedFields["referrer_policy"] = SPEC_JSON["referrer_policy_schema"];
+
+    for (var field in expectedFields) {
+      assert_own_property(scenario, field,
+                          "The scenario contains field " + field)
+      assert_in_array(scenario[field], expectedFields[field],
+                      "Scenario's " + field + " is one of: " +
+                      expectedFields[field].join(", ")) + "."
+    }
+
+    // Check if the protocol is matched.
+    assert_equals(scenario["source_protocol"] + ":", location.protocol,
+                  "Protocol of the test page should match the scenario.")
+
+  }, "[ReferrerPolicyTestCase] The test scenario is valid.");
+
+  var subresourceInvoker = {
+    "a-tag": queryLink,
+    "area-tag": queryAreaLink,
+    "fetch-request": queryFetch,
+    "iframe-tag": queryIframe,
+    "img-tag":  queryImage,
+    "script-tag": queryScript,
+    "worker-request": queryWorker,
+    "xhr-request": queryXhr
+  };
+
+  var pathForSubresource = SPEC_JSON["subresource_path"];
```

Comment:
I'd suggest hard-coding this parameter into the generated `scenario` rather than reading the `spec_json.js` file in every test.